nodebuilder/state/module.go

```diff
@@ -24,9 +24,9 @@ func ConstructModule(tp node.Type, cfg *Config, coreCfg *core.Config) fx.Option
 
 	baseComponents := fx.Options(
 		fx.Supply(*cfg),
+		fx.Supply(cfg.GranterAddress),
```

Comment:
Isn't this just a string? How does this supply work? Can't you just read from the cfg wherever you need it?

Reply:
Yeah, it's a string and It works 🙂 . We don't pass a state cfg to the core accessor, so I can't do it directly. But, as you suggested, I will use an option pattern